packages/x-data-grid/src/tests/keyboard.DataGrid.test.tsx

```diff
@@ -499,96 +499,100 @@ describe('<DataGrid /> - Keyboard', () => {
       );
     }
 
-    it('should scroll horizontally when navigating between column group headers with arrows', function test() {
+    it('should scroll horizontally when navigating between column group headers with arrows', async function test() {
       if (isJSDOM) {
         // Need layouting for column virtualization
         this.skip();
       }
-      render(
-        <div style={{ width: 60, height: 300 }}>
-          <DataGrid
-            columnGroupingModel={columnGroupingModel}
-            autoHeight={isJSDOM}
-            {...getBasicGridData(10, 10)}
-          />
+      const { user } = render(
+        <div style={{ width: 100, height: 300 }}>
+          <DataGrid columnGroupingModel={columnGroupingModel} {...getBasicGridData(10, 10)} />
         </div>,
       );
-      act(() => getColumnHeaderCell(0, 0).focus());
+      // Tab to the first column header
+      await user.keyboard('{Tab}');
       const virtualScroller = document.querySelector<HTMLElement>('.MuiDataGrid-virtualScroller')!;
       expect(virtualScroller.scrollLeft).to.equal(0);
-      fireEvent.keyDown(document.activeElement!, { key: 'ArrowRight' });
+      // We then need to move up to the group header, then right to the first named column
+      await user.keyboard('{ArrowUp}{ArrowUp}{ArrowRight}');
```

Comment:
🤔 It should probably work fine. 

My main grip is that using `.focus()` is not really how the user uses the keyboard. So if you want to test that your behaviour is working properly, you should emulate the entire action.

Reply:
I would leave this call for @mui/xgrid, but maybe we can create an alternative `getColumnHeaderCell` or `focusColumnHeaderCell` function, which would click on the respective cell? 🤔

Reply:
I didn't add clicking on the cell itself cause I don't know if that causes other side effects 

Eg: if we have selectable rows and we click on column 0, then it "selects" the row. 

Focusing through keyboard interactions seemed more safe. 

The main difference on why it has more "keypresses" than before is because the default focus goes to the "effective" first entry on the header if we press `tab`, but these tests are testing the `column groups`, which means they were first focusing on the first ever element on the header `getColumnHeaderCell(0, 0)`, rather than the element that gets the focus first, which in this case would be `getColumnHeaderCell(0, 2)`

So, this means

Previous `getColumnHeaderCell(0, 0)`

| | | |
| --- | --- | --- |
| __*`00`*__ | 10 | 20 | 
| 01 | 11 | 21 |
| 02 | 12 | 22 |

`[Tab]` -> `getColumnHeaderCell(0, 2)`

| | | |
| --- | --- | --- |
| 00 | 10 | 20 | 
| 01 | 11 | 21 |
| __*`02`*__ | 12 | 22 |

I simply moved the focus back to the `getColumnHeaderCell(0, 0)` from the original tests.

Now, it might be that the element at `0,2` shouldn't get focus on tab, which would be another problem unrelated to this PR IMO 😅 

But the tests are comparable between themselves and the assertions didn't change.